roles/openshift_openstack/tasks/check-prerequisites.yml

```diff
@@ -21,17 +21,20 @@
 - name: Check for Neutron trunk support
   os_network_extensions:
 
+# Gather OpenStack service catalog endpoint facts
+- name: Check for cloud service catalog endpoints
+  os_service_catalog:
+
 # Check trunk support
 - fail:
     msg: "Trunk ports enabled but support lacking in Neutron"
   when: (use_trunk_ports | default(False)) and
         ('trunk' not in openstack_network_extensions)
 
-# Check lbaasv2 support
+# Check Kuryr has LBaaSv2 available
 - fail:
-    msg: "Kuryr enabled but lacking required lbaasv2 support in Neutron"
-  when: (openshift_use_kuryr | default(False)) and
-        ('lbaasv2' not in openstack_network_extensions)
+    msg: "Kuryr enabled but OpenStack LBaaS is not."
+  when: openshift_use_kuryr | default(False) and not openshift_openstack_use_lbaas_load_balancer
```

Comment:
I don't think this check is necessary or desired.

We support deployments with single master and infra nodes. In those cases, this forces the creation of 2 load balancers (1 for master and 2 for infra). But only the master/api LB is actually necessary.